File: 1941-check-if-all-characters-have-equal-number-of-occurrences/1941-check-if-all-characters-have-equal-number-of-occurrences.java
class Solution {
    public boolean areOccurrencesEqual(String s) {
        int n = s.length();
        HashMap<Character,Integer> map = new HashMap<>();
        for(int i=0;i<n;i++){
            map.put(s.charAt(i),map.getOrDefault(s.charAt(i),0)+1);
        }
        int ans =map.get(s.charAt(0));
        for(int i=0;i<n;i++){
            if(map.get(s.charAt(i)) != ans){
                return false;
            }
        }
        return true;
        
    }
}

File: 1941-check-if-all-characters-have-equal-number-of-occurrences/1941-check-if-all-characters-have-equal-number-of-occurrences.cpp
class Solution {
public:
    bool areOccurrencesEqual(string s) {
        unordered_map<char,int> map;
        for(char &ch: s){
            map[ch]++;
        }
        int ans =map[s[0]];
        for(char &ch : s){
            if(map[ch] != ans){
                return false;
            }
        }
        return true;
        
    }
};